### PR TITLE
#166539928: improve lucene queries

### PR DESCRIPTION
**Pull request by melniktech**
1. escape quotes;
2. add rule OR for multivalues without special keys, because not it matches by AND rule;
3. add possibility to request without placeholders (if we haven't special keys or multiple values) for avoid breaking changes;
4. correct unit tests

### CODE REVIEW

**Comment by bpneal** [comment]
@melniktech does this need to go into dev too?

**Comment by melniktech** [comment]
> @melniktech does this need to go into dev too?

Yes, it needs. But as I see there are several conflicts to merge in dev (mostly according to versions)

**Comment by melniktech** [comment]
test

File: packages/hiro-graph-lucene/src/index.js [hunk 1]
```diff
@@ -241,18 +256,19 @@ const quote = function(string) {
...
 const slashString = function(input) {
-    return input.replace(/([\\"])/g, function(i, slashOrQuote) {
+    return input.replace(/([\\"'])/g, function(i, slashOrQuote) {
         switch (slashOrQuote) {
             case SLASH:
                 return SLASH + SLASH;
-            case QUOTE:
-                return SLASH + QUOTE;
+            case QUOTE || SINGLE_QUOTE:
+                return `'`;
```
Comment (vteremasov) [inline packages/hiro-graph-lucene/src/index.js]: I don't think we can do so

**Pull request merged**